packages/@aws-cdk/aws-stepfunctions-tasks/lib/emrcontainers/start-job-run.ts

```diff
@@ -0,0 +1,677 @@
+        Name: this.props.jobName,
+        ExecutionRoleArn: this.role.roleArn,
+        ReleaseLabel: this.props.releaseLabel.label,
+        JobDriver: {
+          SparkSubmitJobDriver: {
+            EntryPoint: this.props.jobDriver.sparkSubmitJobDriver?.entryPoint.value,
+            EntryPointArguments: this.props.jobDriver.sparkSubmitJobDriver?.entryPointArguments?.value,
+            SparkSubmitParameters: this.props.jobDriver.sparkSubmitJobDriver?.sparkSubmitParameters,
+          },
+        },
+        ConfigurationOverrides: {
+          ApplicationConfiguration: cdk.listMapper(this.applicationConfigPropertyToJson)(this.props.applicationConfig),
+          MonitoringConfiguration: this.props.monitoring ? {
+            CloudWatchMonitoringConfiguration: this.logGroup ? {
+              LogGroupName: this.logGroup.logGroupName,
+              LogStreamNamePrefix: this.props.monitoring!.logStreamNamePrefix,
+            } : undefined,
+            PersistentAppUI: (this.props.monitoring?.persistentAppUI === false)
+              ? 'DISABLED'
+              : 'ENABLED',
+            S3MonitoringConfiguration: this.logBucket ? {
+              LogUri: this.logBucket.s3UrlForObject(),
+            } : undefined,
+          } : undefined,
+        },
+        Tags: this.renderTags(this.props.tags),
+      }),
+    };
+  }
+
+  /**
+   * Render the EMR Containers ConfigurationProperty as JSON
+   */
+  private applicationConfigPropertyToJson(property: ApplicationConfiguration) {
+    return {
+      Classification: cdk.stringToCloudFormation(property.classification.classificationStatement),
+      Properties: property.properties ? cdk.objectToCloudFormation(property.properties) : undefined,
+      Configurations: property.nestedConfig ? cdk.listMapper(this.applicationConfigPropertyToJson)(property.nestedConfig) : undefined,
+    };
+  }
+
+  private validateAppConfigPropertiesLength(appConfig?: ApplicationConfiguration) {
+    if (appConfig?.properties === undefined) {
+      return;
+    } else if (Object.keys(appConfig.properties).length > 100) {
+      throw new Error(`Application configuration properties must have 100 or fewer entries. Received ${Object.keys(appConfig.properties).length}`);
+    }
+  }
+
+  private validateAppConfigLength(config?: ApplicationConfiguration[]) {
+    if (config === undefined) {
+      return;
+    } else if (config.length > 100) {
+      throw new Error(`Application configuration array must have 100 or fewer entries. Received ${config.length}`);
+    } else {
+      config.forEach(element => this.validateAppConfigLength(element.nestedConfig));
+      config.forEach(element => this.validateAppConfigPropertiesLength(element));
+    }
+  }
+
+  private isArrayOfStrings(value: any): boolean {
+    return Array.isArray(value) && value.every(item => typeof item === 'string');
+  }
+
+  private renderTags(tags?: { [key: string]: any }): { Key: string, Value: string }[] {
```

Comment:
Invalid state machine definition with this current array, the SFN connector expects just an object, not array.

Reply:
if this should just be an object, then we shouldn't be converting into `{ Key: string, Value: string }` in the first place